src/sentry/api/endpoints/project_details.py

```diff
@@ -165,6 +166,33 @@ def validate_relayPiiConfig(self, attrs, source):
             )
         return attrs
 
+    def validate_copy_from_project(self, attrs, source):
+        other_project_id = attrs[source]
+
+        try:
+            other_project = Project.objects.filter(
+                id=other_project_id,
+                organization_id=self.context['project'].organization_id,
+            ).prefetch_related('teams')[0]
+        except IndexError:
+            raise serializers.ValidationError(
+                'Project to copy settings from not found.'
+            )
+
+        request = self.context['request']
+        if not request.access.has_project_access(other_project):
+            raise serializers.ValidationError(
+                'Project settings cannot be copied from a project you do not have access to.'
```

Comment:
Because it appears that members can see all projects and all teams without being a member of them within an organzation, I think these error messages are fine.